### PR TITLE
fix: url validation

### PR DESCRIPTION
**Pull request by czgu**
changed to regex since the previous one would capture json for some reason
also added some unit tests to show its capability

### CODE REVIEW

File: querybook/webapp/ui/Link/Link.tsx [hunk 1]
```diff
@@ -27,6 +27,17 @@ function isInternalUrl(url: LocationDescriptor): boolean {
     return false;
 }
 
+function getExternalUrl(url: LocationDescriptor) {
```
Comment (jczhong84) [inline querybook/webapp/ui/Link/Link.tsx:30]: for external url, whether it starts with `https?` or not, I think it's fine to put it in the `href`.  like `<a href="google.com">google</a>`

or are you trying to solve any other issue?

File: querybook/webapp/lib/utils/index.ts [hunk 2]
```diff
@@ -296,9 +296,9 @@ export function getChangedObject(
     return ret;
 }
 
-// Source: https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url/45567717#45567717
+// Source: https://stackoverflow.com/questions/161738/what-is-the-best-regular-expression-to-check-if-a-string-is-a-valid-url
 const urlPattern =
-    /(?:https?):\/\/(\w+:?\w*)?(\S+)(:\d+)?(\/|\/([\w#!:.?+=&%!\-\/]))?/;
+    /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/i;
```
Comment (jczhong84) [inline querybook/webapp/lib/utils/index.ts:301]: we may simplify `(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?` as `(https?:\/\/)?`, which can also pass all the tests

File: querybook/webapp/ui/Link/Link.tsx (hunk 1 above)
Comment (czgu) [reply querybook/webapp/ui/Link/Link.tsx:30]: that example you put would not redirect to a new site, https://stackoverflow.com/questions/43803778/href-without-https-prefix

File: querybook/webapp/lib/utils/index.ts (hunk 2 above)
Comment (czgu) [reply querybook/webapp/lib/utils/index.ts]: good point

File: querybook/webapp/ui/Link/Link.tsx (hunk 1 above)
Comment (jczhong84) [reply querybook/webapp/ui/Link/Link.tsx:30]: you're right. seems prepending `//` is a better option. also can we merge the check as 
```
if (url && typeof url === 'string' && !/^[a-zA-Z]+:\/\//.test(url)) {
  return `//${url}`
}

return url
```

**Review by jczhong84 — APPROVED** [review]

**Pull request merged**